### PR TITLE
Pootle fs cleanups/tests

### PR DESCRIPTION
**Pull request by phlax**
- adds a path matcher
- FSPlugin delegates equality to wrapped plugin
- cleanups

### CODE REVIEW

File: tests/pootle_fs/files.py [hunk 1]
```diff
@@ -136,6 +136,22 @@ def test_wrap_store_fs_add(store_fs_file_store):
 
 
 @pytest.mark.django_db
+def test_wrap_store_fs_merge_pootle(store_fs_file_store):
+    fs_file = store_fs_file_store
+    fs_file.merge(True)
```
Comment (unho) [inline tests/pootle_fs/files.py]: Just for clarity I would use the named argument `pootle_wins`.

**Comment by unho** [comment]
lgtm.

File: pootle/apps/pootle_fs/matcher.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return self.context.project
+
+    @property
+    def root_directory(self):
+        return self.context.project.local_fs_path
+
+    @property
+    def translation_path(self):
+        return self.context.project.config[
+            "pootle_fs.translation_paths"]["default"]
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/matcher.py]: does "default" translation path limit multiple translation_paths here?

File: pootle/apps/pootle_fs/matcher.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            ["", language_code,
+             self.project.code]
+            + [m for m in
+               matched.get('dir_path', '').split("/")
+               if m]
+            + ["%s.%s"
+               % (filename, ext)])
+
+    def match_pootle_path(self, pootle_path_match=None, **matched):
+        pootle_path = self.make_pootle_path(**matched)
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/matcher.py]: why do we combine matching by `pootle_path_match` and paths codes from `matched`?
when do we use this match?

Comment (phlax) [inline pootle/apps/pootle_fs/matcher.py]: its for reverse matching - ie given a set of lang/subdirs/filename would our `translation_path` give us a fs path, given the fs/pootle glob filter

File: pootle/apps/pootle_fs/matcher.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/pootle_fs/matcher.py]: in the original implementation there was an additional method, this iterated through the translation_path matches using a slightly iffy (but maybe ok) algorithm. This was what i removed.

So now we still have a `translation_paths` dictionary, im thinking we might want to just replace that with a `translation_path` string

Comment (phlax) [inline pootle/apps/pootle_fs/matcher.py]: ...so answering your q, yep this implementation only uses a single translation path - so in a sense limits it, but it doesnt limit from being implemented in the future.

File: pootle/apps/pootle_fs/files.py [hunk 4]
```diff
@@ -133,6 +133,14 @@ def fetch(self):
         self.store_fs.resolve_conflict = FILE_WINS
         self.store_fs.save()
 
+    def merge(self, pootle_wins):
+        if pootle_wins:
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/files.py]: I seems you added tests but I can't understand why it is not covered...

Comment (phlax) [inline pootle/apps/pootle_fs/files.py]: i think it is

**Comment by unho** [comment]
lgtm, nits aside.

**Comment by unho** [comment]
lgtm

File: pootle/apps/pootle_fs/files.py [hunk 5]
```diff
@@ -133,6 +133,14 @@ def fetch(self):
         self.store_fs.resolve_conflict = FILE_WINS
         self.store_fs.save()
 
+    def merge(self, pootle_wins):
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/files.py]: here

**Comment by ta2-1** [comment]
lgtm

**Pull request merged**